From c6b74fa29d709c545e90ac0f8a254d78ba9fe7b2 Mon Sep 17 00:00:00 2001
From: Kyle Fuller <inbox@kylefuller.co.uk>
Date: Sun, 4 Sep 2011 19:18:26 +0000
Subject: [PATCH] Fix CChan::TrimBuffer() (#63)

CChan::TrimBuffer() was trying to do "m_vsBuffer.begin() + -<some number>" which would cause a segmentation fault

Fixes #63
---
 Chan.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Chan.cpp b/Chan.cpp
index b6eb9d4..31e74c0 100644
--- a/Chan.cpp
+++ b/Chan.cpp
@@ -535,7 +535,7 @@ void CChan::ClearBuffer() {
 
 void CChan::TrimBuffer(const unsigned int uMax) {
 	if (m_vsBuffer.size() > uMax) {
-		m_vsBuffer.erase(m_vsBuffer.begin(), m_vsBuffer.begin() + (uMax - m_vsBuffer.size()));
+		m_vsBuffer.erase(m_vsBuffer.begin(), m_vsBuffer.begin() + (m_vsBuffer.size() - uMax));
 	}
 }
 
-- 
1.7.5.4

